### DISCUSSION TITLE
installation help for v3.9

### DISCUSSION THREAD

**Discussion by judevishnu** [glotzerlab/hoomd-blue#1704] [General]
Hi,
I am trying to install hoomd 3.9 in the Mogon cluster. I am having trouble installation where I get the following error:

**CMake Error at /apps/easybuild/core/software/CMake/3.24.3-GCCcore-12.2.0/share/cmake-3.24/Modules/CMakeDetermineCUDACompiler.cmake:277 (message):
  CMAKE_CUDA_ARCHITECTURES must be non-empty if set.
Call Stack (most recent call first):
  CMake/hoomd/HOOMDHIPSetup.cmake:14 (ENABLE_LANGUAGE)
  CMakeLists.txt:110 (include)**

Where and how should I specify this? I am following the below step for installing version 3.9.

set -e
python3 -m venv vhoomdmine --system-site-packages
. vhoomdmine/bin/activate

git clone --recursive --branch v3.9.0  https://github.com/glotzerlab/hoomd-blue.git
python3 /home/jvishnu/hoomd-blue/install-prereq-headers.py

cd /home/jvishnu/hoomd-blue/

mkdir build

cd build/

cmake ..  -DCMAKE_C_FLAGS=-march=native -DENABLE_GPU=ON -DENABLE_MPI=ON -DENABLE_CUDA=ON  -DENABLE_MPI_CUDA=ON  -DBUILD_DEPRECATED=ON  -DBUILD_METAL=OFF -DBUILD_TESTING=ON  -DCOPY_HEADERS=ON -DCMAKE_INSTALL_PREFIX=/home/jvishnu/vhoomdmine/lib/python3.9/site-packages

make -j16 install

best,
Jude

**Comment by joaander** [comment]
You should build the latest version, 4.4.1. It includes new features and fixes *many* critial bugs that are present in 3.9.0.

Your `cmake` command line sets many variables that are not used by the code. CMake should warn you about these. Find a list of all the [options in the documentation](https://hoomd-blue.readthedocs.io/en/v4.4.1/building.html#configure). When you use the documented option to enable GPU compilation, HOOMD-blue's CMake code will set `CMAKE_CUDA_ARCHITECTURES`.

**Reply by judevishnu** [reply]
Thanks upgrading to 4.4.1 solved some issues with unit-test which I had with 3.9.
Best,
Jude
